homework-submission/react-hyf/react-hyf-client/src/HomeworkWeek3.js

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+
+class HomeworkWeek3 extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      data: null
+    };
+  }
+
+  componentDidMount() {
+    fetch("https://uinames.com/api/?amount=10")
+      .then(results => {
+        return results.json();
+      })
+      .then(data => {
+        let names = data.map(item => {
```

Comment:
Setting `jsx` to the state!
this will be a huge object which may or may not be tracked as state, and it will be cause a very slow application in larger ones?